include/openmc/random_ray/source_region.h

```diff
@@ -81,11 +81,211 @@ struct TallyTask {
   };
 };
 
+// The SourceRegionKey combines a base source region (i.e., a material
+// filled cell instance) with a mesh bin. This key is used as a handle
+// for dynamically discovered source regions when subdividing source
+// regions with meshes.
+class SourceRegionKey {
+public:
+  int64_t base_source_region_id;
+  int64_t mesh_bin;
+  SourceRegionKey() = default;
+  SourceRegionKey(int64_t source_region, int64_t bin)
+    : base_source_region_id(source_region), mesh_bin(bin)
+  {}
+
+  // Equality operator required by the unordered_map
+  bool operator==(const SourceRegionKey& other) const
+  {
+    return base_source_region_id == other.base_source_region_id &&
+           mesh_bin == other.mesh_bin;
+  }
+
+  // Hashing functor required by the unordered_map
+  struct HashFunctor {
+    size_t operator()(const SourceRegionKey& key) const
+    {
+      size_t seed = 0;
+      hash_combine(seed, key.base_source_region_id);
+      hash_combine(seed, key.mesh_bin);
+      return seed;
+    }
+  };
+};
+
+class SourceRegionHandle {
+public:
+  //----------------------------------------------------------------------------
+  // Public Data members
+  int negroups_;
+  bool is_numerical_fp_artifact_ {false};
+  bool is_linear_ {false};
+
+  // Scalar fields
+  int* material_;
```

Comment:
Can you perhaps put a comment referring to the description of these members from another class in case someone is wondering what they are?

Reply:
I've added a comment before defining the fields to point the reader in the right direction.